src/base/bits/bits.h

```diff
@@ -1286,19 +1286,25 @@ inline void BitsBase<T, BT, ST>::bitwise_sxr_const(size_t samt, bool arith, Bits
   // Create a mask for extracting the lowest bamt bits from top
   const auto mamt = bits_per_word() - bamt;
   const auto mask = (T(1) << bamt) - 1;
+  // val_ is an array of unsigned values. If we are working with signed values,
+  // we want the top bits of the top-most word to be filled with ones.
+  // This is only used in the case where hob is set and the top word is not
+  // completely full.
+  T upper_most_word = ((-1 << idx) | val_.back());
```

Comment:
Use const and auto wherever possible. Also, to avoid any confusion related to the size of T, use T's constructor when using constants:
```c++
const auto upper_most_word = ((T(-1) << idx) | val_.back());
```